api/nodes/views.py

```diff
@@ -106,8 +118,132 @@ def get_queryset(self):
         return self.get_node().nodes
 
 
-class NodePointersList(generics.ListAPIView, NodeMixin):
+class NodePointersList(generics.ListCreateAPIView, NodeMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+    )
+
     serializer_class = NodePointersSerializer
 
     def get_queryset(self):
         return self.get_node().nodes_pointer
+
+
+class NodePointerDetail(generics.RetrieveDestroyAPIView, NodeMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+    )
+
+    serializer_class = NodePointersSerializer
+
+    # overrides RetrieveAPIView
+    def get_object(self):
+        pointer_lookup_url_kwarg = 'pointer_id'
+        pointer = get_object_or_404(Pointer, self.kwargs[pointer_lookup_url_kwarg])
+        return pointer
+
+    # overrides DestroyAPIView
+    def perform_destroy(self, instance):
+        user = self.request.user
+        auth = Auth(user)
+        node = self.get_node()
+        pointer = self.get_object()
+        node.rm_pointer(pointer, auth)
+        node.save()
+
+
+class NodeFilesList(generics.ListAPIView, NodeMixin):
+    serializer_class = NodeFilesSerializer
+
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+    )
+
+    def get_valid_self_link_methods(self, user, root_folder=False):
+        valid_methods = {'file': [], 'folder': [], }
+        if user is None:
+            return valid_methods
+
+        permissions = self.get_node().get_permissions(user)
+        if 'read' in permissions:
+            valid_methods['file'].append('GET')
+        if 'write' in permissions:
+            valid_methods['file'].append('POST')
+            valid_methods['file'].append('DELETE')
+            valid_methods['folder'].append('POST')
+            if not root_folder:
+                valid_methods['folder'].append('DELETE')
+
+        return valid_methods
+
+    @staticmethod
+    def get_file_item(item, valid_file_methods, node_id, cookie, obj_args):
```

Comment:
This can be simplified. `node_id` seems unnecessary, since you can compute it from `self.get_node()._primary_key` (and this won't be wasted work, because `node_id` doesn't seem to be used anywhere else in `get_queryset`. Same with `valid_file_methods`.